mcserver/views.py

```diff
@@ -522,45 +714,71 @@ def status(self, request, pk):
     # - creates a new trial with "recording" state
     @action(detail=True)
     def record(self, request, pk):
-        def get_count_from_name(name, base_name):
-            try:
-                count = int(name[len(base_name) + 1:])
-                return count
-            except ValueError:
-                return 0
-        
-        session = Session.objects.get(pk=pk)
+        try:
+            if pk == 'undefined':
+                raise ValueError(_("undefined_uuid"))
+            def get_count_from_name(name, base_name):
```

Comment:
seems the function definition should be outside the try catch. It is fine as is, just a style thing.

Reply:
I agree, moved out of the try-catch block.